cas-server-support-osf/src/main/java/io/cos/cas/adaptors/mongodb/OpenScienceFrameworkAuthenticationHandler.java

```diff
@@ -349,4 +353,51 @@ public void setMongoTemplate(final MongoTemplate mongoTemplate) {
     public boolean supports(final Credential credential) {
         return credential instanceof OpenScienceFrameworkCredential;
     }
+
+    /**
+     * Verify User Status.
+     *
+     * "USER_ACTIVE": authentication succeed
+     * "USER_NOT_CONFIRMED": inform the user that the account is not confirmed and provide a resend confirmation link
+     * "USER_DISABLED": inform the user that the account is disable and that they can contact OSF support
+     * "USER_MERGED", "USER_NOT_CLAIMED" and "UNKNOWN": these is not suppose to happen, ask user to contact OSF support
+     *
+     * @param user the OSF user
+     * @return the user status
+     */
+    private String verifyUserStatus(final OpenScienceFrameworkUser user) {
```

Comment:
I double checked this logic with MattF and Steve.
However, this need to be tweaked a little bit for Non-API Django OSF when I port the code.